oxidation/libparsec/crates/platform_local_db/src/lib.rs

```diff
@@ -22,35 +22,33 @@ pub const LOCAL_DATABASE_MAX_VARIABLE_NUMBER: usize = 999;
 #[derive(Clone)]
 pub struct LocalDatabase {
     #[cfg(feature = "test-utils")]
-    executor: Arc<RwLock<Option<Arc<SqliteExecutor>>>>,
-    #[cfg(not(feature = "test-utils"))]
+    path: String,
     executor: Arc<RwLock<Option<SqliteExecutor>>>,
 }
 
 impl LocalDatabase {
+    #[cfg(feature = "test-utils")]
     pub async fn from_path(path: &str) -> DatabaseResult<Self> {
-        #[cfg(feature = "test-utils")]
-        if let Some(local_db) = test_utils::open_local_db_in_memory(path) {
-            return local_db;
-        }
-
-        let connection = new_sqlite_connection_from_path(path).await?;
-        Self::from_sqlite_connection(connection).await
-    }
-
-    pub async fn new_in_memory() -> DatabaseResult<Self> {
-        let connection = SqliteConnection::establish(":memory:")?;
-
-        Self::from_sqlite_connection(connection).await
+        let executor = match test_utils::maybe_open_sqlite_in_memory(path) {
+            Some(executor) => executor,
+            None => {
+                let conn = new_sqlite_connection_from_path(path).await?;
+                SqliteExecutor::spawn(conn)
+            }
+        };
+        Ok(Self {
+            path: path.to_string(),
+            executor: Arc::new(RwLock::new(Some(executor))),
+        })
     }
```

Comment:
@touilleMan We could have keep `new_in_memory`, that would prevent adding `rstest` + `tests_fixtures` in the devDeps

Reply:
I think it's better not to use the in memory mock when testing this crate:
- we are testing the local database (and not using it as part of a higher level component testing) so it's better to test what is going to run in production
-  the perf impact is minimal since the test is very simple